onnxruntime/test/providers/coreml/coreml_basic_test.cc

```diff
@@ -121,17 +119,49 @@ TEST(CoreMLExecutionProviderTest, ArgMaxCastTest) {
   feeds.insert(std::make_pair("X", ml_value_x));
 
   RunAndVerifyOutputsWithEP(model_file_name, "CoreMLExecutionProviderTest.ArgMaxCastTest",
-                            std::make_unique<CoreMLExecutionProvider>(s_coreml_flags),
+                            MakeCoreMLExecutionProvider(),
                             feeds);
 #else
-  // test load only
-  SessionOptions so;
-  InferenceSessionWrapper session_object{so, GetEnvironment()};
-  ASSERT_STATUS_OK(session_object.RegisterExecutionProvider(std::make_unique<CoreMLExecutionProvider>(0)));
-  ASSERT_STATUS_OK(session_object.Load(model_file_name));
-  ASSERT_STATUS_OK(session_object.Initialize());
-  ASSERT_GT(CountAssignedNodes(session_object.GetGraph(), kCoreMLExecutionProvider), 0)
-      << "Some nodes should have been taken by the CoreML EP";
+  TestModelLoad(model_file_name, MakeCoreMLExecutionProvider(), ExpectedEPNodeAssignment::Some);
+#endif
+}
+
+TEST(CoreMLExecutionProviderTest, GatherWithScalarIndices) {
+  // For scalar inputs, the input shape is modified from [] -> [1] before passing the input to CoreML.
+  // This won't work for Gather because the output shape depends on the `indices` input shape which could be a scalar.
+  // Currently, we expect the CoreML EP to only take the Shape node in this graph (Gather -> Shape).
+  const auto model_file_name = ORT_TSTR("testdata/gather_with_scalar_indices_then_shape.onnx");
+
+#if defined(__APPLE__)
+  RandomValueGenerator gen{1234};
+  std::vector<int64_t> X_shape = {5, 3, 4};
+  std::vector<float> X_data = gen.Uniform<float>(X_shape, 0.0f, 1.0f);
+  OrtValue X = CreateInputOrtValueOnCPU<float>(X_shape, X_data);
+  OrtValue indices = CreateInputOrtValueOnCPU<int64_t>(AsSpan<int64_t>({}), AsSpan<int64_t>({1}));
+
+  RunAndVerifyOutputsWithEP(model_file_name, CurrentTestName(),
+                            MakeCoreMLExecutionProvider(),
+                            {{"X", X}, {"indices", indices}});
+#else
+  TestModelLoad(model_file_name, MakeCoreMLExecutionProvider(), ExpectedEPNodeAssignment::Some);
+#endif
+}
+
+TEST(CoreMLExecutionProviderTest, ShapeThenSliceAndGather) {
+  const auto model_file_name = ORT_TSTR("testdata/shape_then_slice_and_gather.onnx");
```

Comment:
nit: comment saying what we're testing here would be good. is it that this is the expected usage we want to validate or is there something special about this combination of ops?